packages/ui-extensions-react/package.json

```diff
@@ -38,7 +38,7 @@
     },
     "peerDependencies": {
         "@datadog/ui-extensions-sdk": "0.30.1",
-        "react": "^17.0.2"
+        "react": "^18.0.2"
```

Comment:
Since we're updating the peer dependencies, we're going to want to give this package a minor bump (minor only because we're still `0.x.y` on the `@datadog/ui-extensions-*` packages). And since this package is getting a bump, we'll also need to bump `@datadog/ui-extensions-sdk`. CI won't pass unless they're both bumped, so it's not really something you should be able to mess up, more a heads up.

Not sure what your familiarity is with `changesets`, but you can either add another changeset with `yarn changeset add` selecting these two packages or edit the front matter in the changeset in this PR: `.changeset/cuddly-pugs-hammer.md`. I'll make a suggestion for the latter, but feel free to ignore it and do the former if you prefer.

Also if you'd prefer to avoid this whole thing, just merge it now and we'll follow-up on the version bumps before making the next release.